src/btorsat.c

```diff
@@ -387,7 +399,10 @@ btor_sat_check_sat (BtorSATMgr *smgr, int32_t limit)
             limit);
   assert (!smgr->satcalls || smgr->inc_required);
   smgr->satcalls++;
-  setterm (smgr);
```

Comment:
This is not the right place to set the terminate callback. Here it only sets the callback for the SAT solver, but if e.g. Boolector already spends all the time in e.g. preprocessing, it should immediately terminate. I would set the callback with `btor_set_term` in the beginning of `btor_check_sat` in `btorcore.c`.